docs/advanced/immutable-actions.md

```diff
@@ -0,0 +1,143 @@
+# Immutable Actions
+
+One of the more common techniques in JavaScript is the use of an object to hold configuration values. The object might be accessed as a global or passed around as an argument. For example:
+
+```TS
+@Component({ .. })
+class ArtistComponent {
+
+  public readonly artist: Artist = { 
+     name: "Johnny Cash", 
+     latestAlbum: "American V" 
+  };
+  
+  constuctor(public store: Store) { }
+  
+  public announce() {
+    this.store.dispatch(new Announce(this.artist)); 
+  }
+
+}
+```
+
+```TS
+export interface Artist {
+  name: string;
+  latestAlbum: string;
+}
+
+export class Announce {
+  static type = '[Announce] action';
+  constructor(public artist: Artist) { }
+}
+
+@State<Artist>({
+  name: 'artistState',
+  defaults: { name: null, latestAlbum: null }
+})
+export class ApplicationState {
+
+  @Action(ApplicationUpdate)
+  public set({ getState, setState }, { artist }: ArtistAction) {
+
+    if (artist.name === "Johnny Cash") {
+        console.log("The Man in Black");
+    } else {
+        console.log(artist.name);
+    }
+    
+    setState({ ...application });
+  }
+
+}
+
+// Outputs: "The Man in Black"
+
+// artistState: {
+//     name: "Johnny Cash",
+//     latestAlbum: "American V"
+// }
+console.log(artist);
+```
+
+
+But in either sort of situation, there is a problem. 
+Functions that have access to the configuration object can modify the object, whether intentionally or accidentally. 
+Suppose that you had a coworker modify the announce function above to highlight Elvis rather than Cash, but they mistyped the comparison.
+
+```TS
+@State<Artist>({ .. })
+export class ApplicationState {
+
+  @Action(ApplicationUpdate)
+  public set({ getState, setState }, { artist }: ArtistAction) {
+
+    if (artist.name = "Elvis Presley") {
```

Comment:
`===` instead of `=`

Reply:
![image](https://user-images.githubusercontent.com/12021443/45919671-d6f20f00-bea1-11e8-9e3c-b9b96bac4c95.png)